lib/signex.ex

```diff
@@ -7,6 +7,8 @@ defmodule SignEx do
   @digest_algorithm Algorithm.default_digest()
   @allowed_algorithms Algorithm.allowed_strings()
 
+  @spec sign(body :: term, metadata :: map, keypair :: map)
+    :: {:ok, {metadata :: map, %SignEx.Parameters{}}} | {:error, atom}
   def sign(_body, %{"signature" => _anything}, _keypair) do
```

Comment:
I think we should add a TODO to rename this `sign_headers` and `verify_headers` or similar

Reply:
maybe `verify_request` would be more accurate, since it handles the body too. I can add the TODO.